File: VB/09_Charts/GetValuesAndUnitFromAxis.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Charts
Imports System.IO

Namespace GetValuesAndUnitFromAxis
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			Dim sb As New StringBuilder()

			'Create PPT document and load file
			Dim ppt As New Presentation()
			ppt.LoadFromFile("..\..\..\..\..\..\Data\ChartSample2.pptx")

			'Get chart on the first slide
			Dim Chart As IChart = TryCast(ppt.Slides(0).Shapes(0), IChart)

			'Get unit from primary category axis
			Dim MajorUnit As Single = Chart.PrimaryCategoryAxis.MajorUnit
			Dim type As ChartBaseUnitType = Chart.PrimaryCategoryAxis.MajorUnitScale

			sb.Append(MajorUnit.ToString() & vbCrLf)
			sb.Append(type.ToString() & vbCrLf)


			'Get values from primary value axis
			Dim minValue As Single = Chart.PrimaryValueAxis.MinValue
			Dim maxValue As Single = Chart.PrimaryValueAxis.MaxValue

			sb.Append(minValue.ToString() & vbCrLf)
			sb.Append(maxValue.ToString() & vbCrLf)


			Dim result As String = "GetValuesAndUnitFromAxis_result.txt"
			'Save the document
			File.WriteAllText(result, sb.ToString())

			'Launch the result file
			PPTDocViewer(result)

		End Sub

		Private Sub PPTDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try

		End Sub

	End Class
End Namespace

File: CS/09_Charts/GetValuesAndUnitFromAxis.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Charts;
using System.IO;

namespace GetValuesAndUnitFromAxis
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            StringBuilder sb = new StringBuilder();

            //Create PPT document and load file
            Presentation ppt = new Presentation();
            ppt.LoadFromFile(@"..\..\..\..\..\..\Data\ChartSample2.pptx");

            //Get chart on the first slide
            IChart Chart = ppt.Slides[0].Shapes[0] as IChart;

            //Get unit from primary category axis
            float MajorUnit = Chart.PrimaryCategoryAxis.MajorUnit;
            ChartBaseUnitType type = Chart.PrimaryCategoryAxis.MajorUnitScale;

            sb.Append(MajorUnit.ToString() + "\r\n");
            sb.Append(type.ToString() + "\r\n");


            //Get values from primary value axis
            float minValue = Chart.PrimaryValueAxis.MinValue;
            float maxValue = Chart.PrimaryValueAxis.MaxValue;

            sb.Append(minValue.ToString() + "\r\n");
            sb.Append(maxValue.ToString() + "\r\n");

                       
            String result = "GetValuesAndUnitFromAxis_result.txt";
            //Save the document
            File.WriteAllText(result, sb.ToString());

            //Launch the result file
            PPTDocViewer(result);

        }

        private void PPTDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }        
        
        }

    }
}